src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -234,38 +211,30 @@ public function createAction(Request $request)
      */
     public function editAction($sqlRequestId, Request $request)
     {
-        $requestSqlFormHandler = $this->getRequestSqlFormHandler();
-
-        $editRequestSqlForm = $requestSqlFormHandler->getFormFor($sqlRequestId);
-        $editRequestSqlForm->handleRequest($request);
-
-        if ($editRequestSqlForm->isSubmitted()) {
-            if ($this->isDemoModeEnabled()) {
-                $this->addFlash('error', $this->getDemoModeErrorMessage());
-
-                return $this->redirectToRoute('admin_sql_request');
-            }
-
-            $errors = $requestSqlFormHandler->save($editRequestSqlForm->getData());
+        try {
+            $sqlRequestForm = $this->getSqlRequestFormBuilder()->getFormFor($sqlRequestId);
+            $sqlRequestForm->handleRequest($request);
 
-            if (empty($errors)) {
-                $this->addFlash(
-                    'success',
-                    $this->trans('Successful update.', 'Admin.Notifications.Success')
-                );
+            if ($this->getSqlRequestFormHandler()->handleFor($sqlRequestId, $sqlRequestForm)) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
 
-                return $this->redirectToRoute('admin_sql_request');
+                return $this->redirectToRoute('admin_sql_requests_index');
             }
+        } catch (SqlRequestNotFoundException $e) {
```

Comment:
`SqlRequestNotFoundException` it is `Domain` exception, so i dont think that `IdentifiableObjectNotFound` could be a parent of `SqlRequestNotFoundException` since its a separate application layers. wdyt?

Reply:
we could have genereic basic exceptions; like `DatabaseException`, 
 `ResultNotFoundException`, ... which can be used as parents for domain specific exceptions
this will help us generalize the behavior of the controllers later, and for domain specific actions it still leaves the possibility to catch the domain specific methods

Reply:
actually some basic exceptions already exist: https://github.com/PrestaShop/PrestaShop/blob/1.7.5.x/src/Core/Exception/DatabaseException.php